spec/blocksync/readme.md

```diff
@@ -0,0 +1,64 @@
+---
+order: 
+parent:
+    title: blocksync
+    order: 
+---
+
+
+In a proof of work blockchain, syncing with the chain is the same process as staying up-to-date with the consensus: download blocks, and look for the one with the most total work. In proof-of-stake, the consensus process is more complex, as it involves rounds of communication between the nodes to determine what block should be committed next. Using this process to sync up with the blockchain from scratch can take a very long time. It's much faster to just download blocks and check the merkle tree of validators than to run the real-time consensus gossip protocol.
```

Comment:
But more generally, I don't agree with its content. In both classes of blockchains, syncing blocks is the same as deciding the block in the usual (consensus) way. In PoW blockchains, we extend a block that we "trust" with downloaded blocks, verify the downloaded block, and append to a local candidate blockchain. If we have multiple local candidate blockchains, we opt for the longest one, that represents the greatest accumulate work. In PoS blockchains, we do the same: we extend a block that we "trust" with downloaded blocks, verify the downloaded block, and append to a local candidate blockchain. The differences are two: (i) we verify a block by verifying the signatures of messages voting for the block, instead of verifying brute-force-generated hashes, and (ii) we should not have multiple candidate blockchains.

That said, one thing that should be clearer in the specification of this module/protocol/spec is that blocksync represent a very condensed execution of the consensus protocol. The consensus protocol involves (potentially) multiples rounds, each round is composed by three communication steps. A node, however, can decide a block by only receiving the messages from the first (propose) and last (precommit) step of any round, provided they match. In other words, it is enough for a node to receive a proposed block and `2/3+ Precommit` messages from the same round for that block to complete and instance of consensus.

In my view, and please correct me if I am wrong, this exactly the information used in blocksync. There are, of course, differences: (i) the proposed block is not split into multiple messages, as in consensus, and (ii) a set of `2/3+` identical `Precommit` messages for the decided block are condensed into a `Commit`. If this parallel between consensus and blocksync is indeed valid, I think that this is the way of introducing blocksync.